### DISCUSSION TITLE
Install help

### DISCUSSION THREAD

**Discussion by Applesupnorth** [grblHAL/core#154] [General]
Hey all, I'm having some trouble getting grblHAL/LPC176x setup so I can compile for a MKS SGen L.  I can't seem to get the libraries added correctly so that I can compile/upload with Arduino IDE.  I've followed the Wiki, but to no avail...  Can someone point me in the right direction and maybe show me a screenshot of the directory structure in your arduino/libraries folder.?.?. Thanks for any help!!!

**Comment by terjeio** [comment]
This driver is built with MCUExpresso v11, I do not know how much work it will be to convert it to a [Arduino project.](https://github.com/grblHAL/LPC176x/issues/27)

**Reply by Applesupnorth** [reply]
Sorry, I'm very much a rookie when it comes to such things... I'll check out MCUExpresso(whatever that is :P lol)   Just wondering why the grblHAL wiki would have a section on compiling with Arduino IDE??

**Reply by dresco** [reply]
> Just wondering why the grblHAL wiki would have a section on compiling with Arduino IDE??

It's because the core wiki tries to cover multiple platforms/drivers, which are built with different tools.. 

I've never built the LPC drivers,  but I believe the NXP MCUExpresso IDE is based on Eclipse - so the [Eclipse](https://github.com/grblHAL/core/wiki/Compiling-GrblHAL#eclipse-based-ide) section in the Wiki may be more help?

**Reply by Applesupnorth** [reply]
That makes more sense now thank you!  I'll check out Eclispe as well!  I have a new board coming in, so the necessity of using Sgen L board with the LPC cpu will be gone, but I'd still like to learn about the process for the future.  Thanks again
